### ISSUE TITLE
@dataProvider with JUnit logs does not log class attribute to files

### ISSUE THREAD

**Issue by tarjei** [sebastianbergmann/phpunit#1952]
This causes Jenkins xUnit PhpUnit parsing to fail.

How hard is it to fix this?

**Comment by sebastianbergmann** [comment]
Can you please be more specific? Thanks.

**Comment by tarjei** [comment]
No problem, apologizes for a bad bugreport. 

My initial problem was that when I started using ParaTest to run my tests in parallel with --log-junit then Jenkins reported parse failures on the files. After some studying of the problem I thought I understood the error and that it was that tests with a dataprovider reported the testresults without a class attribute on the test like this: 
```
 <testcase name="testCourseAccessCombinations with data set &quot;no sphere, course&quot;" assertions="4" time="0.000511"/>
```
I then noticed that PHPUnit also had the same behaviour and (incorrectly) thought that Paratest was using PHPUnits logging infrastructure. 

I now see that the problem was that ParaTest was reporting empty attributes for the line, class and file values on these testcases whereas PHPUnit doesn't report them at all. It would be preferable if it did, but that is not the issue here. 

Anyhow, I'm sorry for waisting your time, thanks for responding.
